server/etcdserver/server.go

```diff
@@ -2075,16 +2074,12 @@ func (s *EtcdServer) snapshot(snapi uint64, confState raftpb.ConfState) {
 	// So KV().Commit() cannot run in parallel with toApply. It has to be called outside
 	// the go routine created below.
 	s.KV().Commit()
+	d := GetMembershipInfoInV2Format(s.Logger(), s.cluster)
```

Comment:
Overall note about code changes. Please avoid unnecessary reordering between functions in PR, it adds additional review strain. Like in here between `s.KV().Commit()`, `GetMembershipInfoInV2Format(s.Logger(), s.cluster)`. I don't think it matters here, but takes a minute or two to think about this.

Reply:
Created https://github.com/etcd-io/etcd/pull/16460 to keep the order same to simplify reasoning / avoid the unnecessary strain.